mqtt.py

```diff
@@ -24,13 +26,63 @@ def stream(self, detect_request: str = '{}'):
             # Run the stream detector and return the results.
             streamer = Streamer(self.doods).start_stream(detect_request)
             for detect_response in streamer:
-                # If we requested an image, base64 encode it back to the user
-                if detect_request.image:
-                    detect_response.image = base64.b64encode(detect_response.image).decode('utf-8')
-                for detection in detect_response.detections:
+                # If separate_detections, iterate over each detection and process it separately
+                if detect_request.separate_detections:
+
+                    #If we're going to be cropping, do this processing only once (rather than for each detection)
+                    if detect_request.image and detect_request.crop:
+                        detect_image_bytes = np.frombuffer(detect_response.image, dtype=np.uint8)
+                        detect_image = cv2.imdecode(detect_image_bytes, cv2.IMREAD_COLOR)
+                        di_height, di_width = detect_image.shape[:2]
+
+
+                    for detection in detect_response.detections:
+                        detection_dict = detection.asdict(include_none=False)
```

Comment:
Can't you just assign to the field that's already in the dataclass object? Does it need to be a dict?

Reply:
If the detections are each being returned separately, the base64 image needs to be returned in the `detection` object (on line 61) but there's no image field.  Since it's being `asdict`ed anyway for the return, this allowed me to add an image field without modifying the dataclass (since in the API there won't be images here).

As long as you don't mind adding an `image` field to the dataclass, I can just put the image there.  Let me know.

Reply:
Yeah, just add it there. I think that makes enough sense.

Reply:
Done.